diff --git a/PKGBUILD b/PKGBUILD
index ecbbdbe..b2942b7 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -53,7 +53,7 @@ package_intellij-idea-ultimate-edition() {
 
   cd "$srcdir"
 
-  install -d "$pkgdir"/{opt/$pkgname,usr/bin}
+  install -d "$pkgdir"/{opt/$pkgname,usr/bin,usr/share/icons/hicolor/scalable/apps}
   mv idea-IU-${_buildver}/* "$pkgdir"/opt/$pkgbase
 
   # https://youtrack.jetbrains.com/issue/IDEA-185828
@@ -61,7 +61,7 @@ package_intellij-idea-ultimate-edition() {
 
   ln -s /opt/$pkgname/bin/idea "$pkgdir"/usr/bin/$pkgname
   install -D -m644 "$srcdir"/jetbrains-idea.desktop "$pkgdir"/usr/share/applications/jetbrains-idea.desktop
-  install -D -m644 "$pkgdir"/opt/$pkgbase/bin/idea.svg "$pkgdir"/usr/share/pixmaps/"$pkgname".svg
+  ln -s /opt/$pkgbase/bin/idea.svg "$pkgdir"/usr/share/icons/hicolor/scalable/apps/"$pkgname".svg
 
   # workaround FS#40934
   sed -i 's|lcd|on|'  "$pkgdir"/opt/$pkgname/bin/*.vmoptions
