diff --git a/PKGBUILD b/PKGBUILD
index 8f3cc5c..b2f753e 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -52,7 +52,6 @@ package() {
 		lang=$(sed -e 's/.mo$//' <<< "${langmo}")
 		install -Dm644 "locale/${langmo}" "$pkgdir/usr/share/locale/${lang}/LC_MESSAGES/pikaur.mo"
 	done
-	install -Dm644 LICENSE "${pkgdir}/usr/share/licenses/${pkgname}/LICENSE"
 	install -Dm644 pikaur.1 "${pkgdir}/usr/share/man/man1/${_pkgname}.1"
 	cp -r ./packaging/* "${pkgdir}"
 }
